crates/astria-sequencer-relayer/src/relayer/write/conversion.rs

```diff
@@ -92,3 +120,22 @@ pub(super) fn convert(block: SequencerBlock) -> eyre::Result<Converted> {
         },
     })
 }
+
+fn brotli_compressed_bytes(data: &[u8]) -> eyre::Result<Vec<u8>> {
+    use std::io::Write as _;
+    let compression_params = BrotliEncoderParams {
+        quality: 5,
+        size_hint: data.len(),
+        ..Default::default()
+    };
+    let mut output = Vec::new();
```

Comment:
An easy perf improvement might be `Vec::with_capacity(data.len())` (as an upper limit) or (maybe better because deterministic) a multiple of the buffer size set in `CompressorWriter::with_params`: i.e. `Vec::with_capacity(BUF_SIZE * 8)` with `BUF_SIZE = 4096`.

`Vec` grows exponentially. In it would reallocate like `0 -> 4096 -> 8192 -> 16384` (note that `Vec::new()` is a no-op, so in this case doing `Vec::with_capacity(4096)` or not does the same amount of work).